AppConnectSwift/CaptureImageViewController.swift

```diff
@@ -56,55 +54,29 @@ class CaptureImageViewController: UIViewController, UIImagePickerControllerDeleg
-                
-                subject.collectData(self.data, withMetadata: "Random String", contentType: "image/jpeg", completion: { (dataEnvelope:  MDSubjectDataEnvelope!, err: NSError!) -> Void in
-                    if err == nil {
-                        client.sendEnvelope(dataEnvelope, completion: { (err) in
-                            if err == nil {
-                                NSOperationQueue.mainQueue().addOperationWithBlock {
-                                    self.showAlert("Data saved successfully", message: "")
-                                    self.imageView.image = nil
-                                    subject = nil
-                                    bgQueue = nil
-                                    datastore = nil
-                                }
-                            }
-                            else if err.description.containsString("The Internet connection appears to be offline"){
-                                NSOperationQueue.mainQueue().addOperationWithBlock {
-                                    self.showAlert("Not connected to the Internet", message: "Please restore Internet connection and try again")
-                                    bgQueue = nil
-                                    datastore = nil
-                                }
-                            }
-                            else {
-                                NSOperationQueue.mainQueue().addOperationWithBlock {
-                                    self.showAlert("Unable to save data", message: err.description)
-                                    bgQueue = nil
-                                    datastore = nil
-                                }
-                            }
-                        })
-                    }
-                    else{
-                        self.showAlert("", message: err.description)
-                        bgQueue = nil
-                        datastore = nil
-                        subject = nil
-                    }
-                })
-            }
-        }
-        else {
+        
+        if self.data == nil {
             showAlert("No image selected", message: "")
+            return
         }
+        
+        let datastore = (UIApplication.sharedApplication().delegate as! AppDelegate).UIDatastore!
+        let subject = datastore.subjectWithID(self.subjectID)
+        
+        subject.collectData(self.data, withMetadata: "Random String", contentType: "image/jpeg", completion: { (err: NSError!) -> Void in
+            if (err == nil) {
+                
+                // update the UI.
+                dispatch_async(dispatch_get_main_queue()) {
+                    self.showAlert("Data saved successfully", message: "Will be uploaded automatically!")
+                    self.imageView.image = nil
+                    self.data = nil
+                    self.saveImageButton.enabled = false;
+                }
+            }
```

Comment:
We might. This was originally just "ideal path demo" but I could certainly see adding more. There aren't a lot of scenarios where an immediate error would be very informative though, as this is simply the call that will enqueue the data for upload, and from there errors are handled (and retried) all behind the scenes, with the outcome unfolding in the logs.